src/ClientWidgetApi.ts

```diff
@@ -550,6 +558,40 @@ export class ClientWidgetApi extends EventEmitter {
         });
     }
 
+    private handleUpdateDelayedEvent(request: IUpdateDelayedEventFromWidgetActionRequest) {
+        if (!request.data.delay_id) {
+            return this.transport.reply<IWidgetApiErrorResponseData>(request, {
+                error: {message: "Invalid request - missing delay_id"},
+            });
+        }
+
+        switch (request.data.action) {
+            case UpdateDelayedEventAction.Cancel:
+            case UpdateDelayedEventAction.Restart:
+            case UpdateDelayedEventAction.Send:
+                break
+            default:
+                return this.transport.reply<IWidgetApiErrorResponseData>(request, {
+                    error: {message: "Invalid request - unsupported action"},
+                });
+        }
+
+        if (!this.hasCapability(MatrixCapabilities.MSC4157UpdateDelayedEvent)) {
+            return this.transport.reply<IWidgetApiErrorResponseData>(request, {
+                error: {message: "Missing capability"},
+            });
+        }
+
+        this.driver.updateDelayedEvent(request.data.delay_id, request.data.action).then(() => {
+            return this.transport.reply<IWidgetApiAcknowledgeResponseData>(request, {});
+        }).catch(e => {
+            console.error("error updating delayed event: ", e);
+            return this.transport.reply<IWidgetApiErrorResponseData>(request, {
+                error: {message: "Error updating delayed event"},
+            });
+        });
+    }
```

Comment:
I think it also makes sense to first check the capability. Then the main switch becomes more readable (not uses the `break` as a synonym for "I will not return but send"
```suggestion
private handleUpdateDelayedEvent(request: IUpdateDelayedEventFromWidgetActionRequest) {
    if (!request.data.delay_id) {
        return this.transport.reply<IWidgetApiErrorResponseData>(request, {
            error: {message: "Invalid request - missing delay_id"},
        });
    }

    if (!this.hasCapability(MatrixCapabilities.MSC4157UpdateDelayedEvent)) {
        return this.transport.reply<IWidgetApiErrorResponseData>(request, {
            error: {message: "Missing capability"},
        });
    }

    switch (request.data.action) {
        case UpdateDelayedEventAction.Cancel:
        case UpdateDelayedEventAction.Restart:
        case UpdateDelayedEventAction.Send:
            this.driver.updateDelayedEvent(request.data.delay_id, request.data.action).then(() => {
                return this.transport.reply<IWidgetApiAcknowledgeResponseData>(request, {});
            }).catch(e => {
                console.error("error updating delayed event: ", e);
                return this.transport.reply<IWidgetApiErrorResponseData>(request, {
                    error: {message: "Error updating delayed event"},
                });
            });
        default:
            return this.transport.reply<IWidgetApiErrorResponseData>(request, {
                error: {message: "Invalid request - unsupported action"},
            });
    }
}
```